### PR TITLE
cmd: Adds check for invalid inferred IP

### LINKED ISSUE #1829: nil pointer error in getServiceURI() on startup
[Original report](https://github.com/livepeer/go-livepeer/issues/1780#issuecomment-788902131)

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x10c48c5]

goroutine 1 [running]:
net/url.(*URL).Hostname(...)
        /usr/local/go/src/net/url/url.go:1112
main.getServiceURI(0xc0000dc160, 0x0, 0x0, 0x0, 0x0, 0x0)
        /build/cmd/livepeer/livepeer.go:1070 +0x2e5
main.main()
        /build/cmd/livepeer/livepeer.go:873 +0x5185
```

Reported on v0.5.14.

### PR DESCRIPTION
**Pull request by reubenr0d**
**What does this pull request do? Explain your changes. (required)**

**Specific updates (required)**

- Adds check for invalid inferred IP for when `https://api.ipify.org?format=text` responds with success but there is an issue in the response

**How did you test each of these updates (required)**

- Hardcoded the response from IP check as "%" using `body = []byte("%")` to reproduce the error

**Does this pull request close any open issues?**

Fixes #1829

**Checklist:**

- [x] Read the [contribution guide](./doc/contributing.md)
- [x] `make` runs successfully
- [x] All tests in `./test.sh` pass
- [x] README and other documentation updated
- [x] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: cmd/livepeer/livepeer.go [hunk 1]
```diff
@@ -1018,6 +1018,10 @@ func getServiceURI(n *core.LivepeerNode, serviceAddr string) (*url.URL, error) {
 	}
 	addr := "https://" + strings.TrimSpace(string(body)) + ":" + RpcPort
 	inferredUri, err := url.ParseRequestURI(addr)
+	if err != nil {
+		glog.Error("Could not look up public IP address")
```
Comment (kyriediculous) [inline cmd/livepeer/livepeer.go]: Let's include the original error message here as well: 

`glog.Errorf("Could not look up public IP err=%v", err)`

File: CHANGELOG_PENDING.md [hunk 2]
```diff
@@ -21,6 +21,8 @@
 
 #### General
 
+- \#1829 Adds check for invalid inferred IP (@reubenr0d)
```
Comment (kyriediculous) [inline CHANGELOG_PENDING.md]: Let's change this to 1859, the number of this PR

**Review by kyriediculous — CHANGES_REQUESTED** [review]

File: cmd/livepeer/livepeer.go (hunk 1 above)
Comment (reubenr0d) [reply cmd/livepeer/livepeer.go]: Sure, let me add it to the other error messages here as well.

File: CHANGELOG_PENDING.md [hunk 3]
```diff
@@ -21,6 +21,8 @@
 
 #### General
 
+- \#1859 Adds check for invalid inferred IP (@reubenr0d)
```
Comment (kyriediculous) [inline CHANGELOG_PENDING.md]: Looks good , just one small detail here. This change only impacts the orchestrator so let's change this to be under "Orchestrator"

Perhaps we could also update the text to be a little more descriptive, maybe something like "Handle error for invalid inferred orchestrator public IP address on node startup"

**Review by kyriediculous — CHANGES_REQUESTED** [review]

**Comment by kyriediculous** [comment]
There appears to be a merge conflict, you should be able to pull the latest upstream master and rebase on top of it. Once that's done should be GTG.

**Review by kyriediculous — APPROVED** [review]
LGTM 👍

**Comment by kyriediculous** [comment]
I forgot to mention, let's keep the PR open until a bounty is attached

**Pull request merged**